java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -62,14 +68,33 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
       case INSTANCE_OF:
         result = new SymbolicValue.InstanceOfSymbolicValue(counter);
         break;
+      case MEMBER_SELECT:
+        result = createIdentifierSymbolicValue(((MemberSelectExpressionTree) syntaxNode).identifier());
+        break;
+      case IDENTIFIER:
+        result = createIdentifierSymbolicValue((IdentifierTree) syntaxNode);
+        break;
       default:
-        result = symbolicValueFactory == null ? new SymbolicValue(counter) : symbolicValueFactory.createSymbolicValue(counter, syntaxNode);
-        symbolicValueFactory = null;
+        result = createDefaultSymbolicValue(syntaxNode);
     }
     counter++;
     return result;
   }
 
+  private SymbolicValue createIdentifierSymbolicValue(IdentifierTree identifier) {
+    if (BOOLEAN_CLASS_VALUES.contains(identifier.name()) && identifier.symbolType() != null && identifier.symbolType().is(BOOLEAN_CLASS)) {
```

Comment:
identifier.symbolType() generates an NPE in testFactory(org.sonar.java.se.SymbolicValueFactoryTest), so one needs to check for null.

The method getConstraintForPrimitiveType() wraps all the necessary cehcking to compatible type, just like the Java primitive Object wrappers are doing. Your alternative would shortcircuit a check for  compatibility. Of course, you can work without any net...